### PR TITLE
net/rdcli*: restructure and rename RD related modules

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
As discussed in #7428: the Resource Directory endpoint implementations that we merged (i.e. `rdcli` and `rdcli_simple`) use some naming conventions that are not in line with the wording used in [draft-ietf-core-resource-directory-15](https://tools.ietf.org/html/draft-ietf-core-resource-directory-15).

This PR restructures all existing resource directory code by applying a more clear, and extensible structure, while following the wording used in the draft.

I introduced a high-level module called `cord`, which includes all the code related to interacting with resource directories. All existing (endpoint) and future (lookup client) code is put as submodules into this high level module. See `sys/net/application_layer/cord/doc.txt` as included in this PR for more information on the high-level structure.

Sorry for the amount of lines of code in this PR, but I think it does make sense to have all these changes in a single PR. I tried to do the changes in self-contained commits to make reviewing as easy as possible...

### Testing procedure
Simply verify that the two example applications (now `cord_ep` and `cord_epsim`) are still working as expected.

Also check the generated doxygen.

### Issues/PRs references
#7428

### CODE REVIEW

File: examples/cord_ep/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+can change these values at compile time by overriding their defines:
+Using command line arguments:
+```
+CFLAGS="-DCORD_EP=\\\"your_ep_name_here\\\"" make all
+```
+Or by setting their values in the application's Makefile:
+```
+CFLAGS += "-DCORD_EP=\"MyNewEpName\""
+```
+
```
Comment (miri64) [inline examples/cord_ep/README.md]: Git is complaining about this whitespace.

File: examples/cord_ep/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+CoRE Resource Directory: Endpoint Example
+=========================================
+
+This example application demonstrates the usage of RIOT's Resource Directory
+(RD) endpoint module, called `cord_ep`. This module supports the registration,
+update, and removal procedures as defined in the
```
Comment (miri64) [inline examples/cord_ep/README.md]: Wording: "in the draft-ietf-core-…" sound's clunky (might be an artifact from the previous version where it was "in the […] draft".

File: examples/cord_ep/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Usage
+=====
+The examples includes a shell command that you can use to interact with a given
+RD, called `cord_ep`. Simply use that shell command without parameters for
+more information on its usage.
+
+Some connection parameters are configured statically during compile time,
+namely the lifetime (`CORD_LT`) and the node's endpoint name (`CORD_EP`). You
+can change these values at compile time by overriding their defines:
+Using command line arguments:
```
Comment (miri64) [inline examples/cord_ep/README.md]: "overriding their defines using command line arguments"

File: sys/include/net/cord/epsim.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ * @return  CORD_EPSIM_ERROR if something goes wrong preparing or sending the
+ *          initial request
+ */
+int cord_epsim_register(void);
+
+#if defined(MODULE_CORD_EPSIM_STANDALONE) || defined(DOXYGEN)
+/**
+ * @brief   Spawn a new thread that registers the node and updates the
+ *          registration with all responding RDs using the simple registration
+ *          process
```
Comment (miri64) [inline sys/include/net/cord/epsim.h]: The `#if defined(MODULE_CORD_EPSIM_STANDALONE)` is new and should be documented:

```C
/**
 […]
 * @note   Only available with @ref net_cord_ep_standalone compiled in.
 */
```

Comment (haukepetersen) [reply sys/include/net/cord/epsim.h]: added

**Comment by haukepetersen** [comment]
addressed comments and rebased.

File: sys/include/net/cord/epsim.h [hunk 5]
```diff
@@ -51,6 +51,10 @@ int cord_epsim_register(void);
  * @brief   Spawn a new thread that registers the node and updates the
  *          registration with all responding RDs using the simple registration
  *          process
+ *
+ * @note    Only available with @ref cord_epsim_standalone compiled in
+ * @note    This function must only be called once (typically during system
```
Comment (miri64) [inline sys/include/net/cord/epsim.h]: I think this should rather be a `@warning` or `@attention` even.

Comment (haukepetersen) [reply sys/include/net/cord/epsim.h]: @warning seems sensible. Will also adapt that in `cord/ep_standalone.h`.

**Comment by haukepetersen** [comment]
addressed last comment.

File: sys/include/net/cord/epsim.h [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ */
+int cord_epsim_register(void);
+
+#if defined(MODULE_CORD_EPSIM_STANDALONE) || defined(DOXYGEN)
+/**
+ * @brief   Spawn a new thread that registers the node and updates the
+ *          registration with all responding RDs using the simple registration
+ *          process
+ *
+ * @note    Only available with @ref cord_epsim_standalone compiled in
```
Comment (miri64) [inline sys/include/net/cord/epsim.h]: *`net_`*`cord_epsim_standalone` ;-)

**Comment by miri64** [comment]
You can squash immediately.

**Comment by haukepetersen** [comment]
done

**Pull request merged**